snuba/datasets/profiles_processor.py

```diff
@@ -9,44 +9,80 @@
 
 metrics = MetricsWrapper(environment.metrics, "profiles.processor")
 
-RETENTION_DAYS_ALLOWED = frozenset([30, 90])
-
 
 class ProfilesMessageProcessor(MessageProcessor):
     def process_message(
         self, message: Mapping[str, Any], metadata: KafkaMessageMetadata
     ) -> Optional[ProcessedMessage]:
         try:
-            retention_days = message["retention_days"]
-            if retention_days not in RETENTION_DAYS_ALLOWED:
-                retention_days = 30
+            if "version" in message:
+                additional_fields = {
+                    f"device_{field}": message["device"].get(field, "")
+                    for field in [
+                        "locale",
+                        "classification",
+                        "manufacturer",
+                        "model",
+                    ]
+                }
+                additional_fields["architecture"] = message["device"].get(
+                    "architecture", "unknown"
+                )
+                additional_fields["version_name"] = message["release"]
+
+                for field in ["name", "version", "build_number"]:
+                    additional_fields[f"device_os_{field}"] = message["os"].get(field)
+
+                transaction = message["transactions"][0]
+                additional_fields.update(
+                    {
+                        "transaction_id": transaction["id"],
```

Comment:
Do we need to format this with
```suggestion
                        "transaction_id": str(UUID(transaction["id"])),
```

Reply:
I mean sure, it'll be the 3rd time in the pipeline we format it but you know.